.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,97 @@
+    if: "!contains(github.event.head_commit.message, '[automated]') && !contains(github.event.head_commit.message, '[no-deploy]')"
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Set deployment type
+        run: |
+          # Check if not triggered from worflow-dispatch (event input variable is empty)
+          if [ -z "${{github.event.inputs.deploy_type}}" ]; then
+            # Set deployment type to release as was not triggered by workflow_dispatch
+            DEPLOY_TYPE="release"
+          else
+            # Set deployment type to workflow_dispatch event input variable
+            DEPLOY_TYPE=${{github.event.inputs.deploy_type}}
+          fi
+          # Store the appName in environment variables based on deployment type
+          if [[ $DEPLOY_TYPE = "test" ]]; then
+            echo "appName=MetaInsight_test" >> $GITHUB_ENV
+          else
+            echo "appName=MetaInsight" >> $GITHUB_ENV
+          fi
+      - name: Print deployment type
+        run: |
+          echo "Deploying to: $appName"
+      - name: Install gcc 11
+        run: |
+          sudo apt update
+          sudo apt install gcc-11 g++-11 -y
+          sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-11 110 --slave /usr/bin/g++ g++ /usr/bin/g++-11 --slave /usr/bin/gcov gcov /usr/bin/gcov-11
+      - name: Install package system dependencies
+        run: |
+          sudo apt install libcurl4-openssl-dev jags libharfbuzz-dev libfribidi-dev libfreetype6-dev libpng-dev libtiff5-dev libjpeg-dev libglpk-dev -y
+      # Checkout the repo
+      - uses: actions/checkout@v4
+        with:
+          ref: "{{github.event.inputs.manual_branch}"
+      # Setup R (4.3.0)
+      - uses: r-lib/actions/setup-r@v2
+        with:
+          r-version: '4.3.2'
+      # Setup and install dependencies using renv
+      - uses: r-lib/actions/setup-renv@v2
+      - name: Check GCC Version
+        run: |
+          gcc --version
+          if ! gcc --version | grep -q 11; then exit 1; fi
+      # Deploy using rsconnect
+      - name: Deploy
+        run: |
+          # Install rsconnect
+          install.packages("rsconnect")
+          # Set up account
+          cat("Checking account info...")
+          rsconnect::setAccountInfo(
+            Sys.getenv("accountName"),
+            Sys.getenv("accountToken"),
+            Sys.getenv("accountSecret")
+          )
+          cat(" [OK]\n")
+          cat(paste0("Deploying ", Sys.getenv("appName"), "\n"))
+          # Deploy application
+          rsconnect::deployApp(
+            appName = Sys.getenv("appName"),
+            account = Sys.getenv("accountName"),
+            forceUpdate = TRUE
+          )
+        shell: Rscript {0}
+        env:
+          accountName: crsu
+          # token and secret obtained from https://www.shinyapps.io/admin/#/tokens
+          accountToken: ${{ secrets.SHINYAPPS_TOKEN }}
```

Comment:
These need to be set in repository secrets before use, settings > secrets and variables > repository secrets